### PR TITLE
Feature/271 e2 e tests sprint 7

### PR DESCRIPTION
**Pull request by Elaine-Krauss-TCG**
Implements E2E tests for the Sprint 7 QA Scripts (#271)

Refactors the E2E tests from Sprint 6 and their support files to meet code style guidelines per the comments on #209's pull request

### CODE REVIEW

File: front-end/cypress/integration/sprint-7-test-119.spec.ts [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+  it("Step 3: Set the Contact Type to 'Candidate'", () => {
+    cy.DropdownSetValue("p-dropdown[FormControlName='type']", 'Candidate');
+    cy.wait(50);
+    cy.get("p-dropdown[FormControlName='type']").should('contain', 'Candidate');
+  });
+
+  it("Steps 4 & 5: Verify 'Candidate Office' dropdown contains House, Senate, and Presidential and then select Presidential", () => {
+    cy.get(FieldCandidateOffice).click();
+
+    let OfficeType: string;
```
Comment (toddlees) [inline front-end/cypress/integration/sprint-7-test-119.spec.ts]: This is causing a code smell. You don't need this variable if it exists fully in the scope of the for-loop below

File: front-end/cypress/integration/sprint-7-test-192,248,249,250.spec.ts [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+  it('Step 1: Navigate to contacts page', () => {
+    cy.visit('/dashboard');
+    cy.url().should('contain', '/dashboard');
+    cy.get('.p-menubar').find('.p-menuitem-link').contains('Contacts').click();
+    cy.url().should('contain', '/contacts');
+  });
+
+  for (ContactType of Object.keys(Contacts)) {
+    context(`---> ${ContactType}`, (c_type = ContactType) => {
+      let Contact: object = Contacts[c_type];
```
Comment (toddlees) [inline front-end/cypress/integration/sprint-7-test-192,248,249,250.spec.ts]: This is an unused variable.  You can remove it

**Review by toddlees — CHANGES_REQUESTED** [review]
There are a couple cognitive complexity code smells.  In the core codebase, they would be a problem because devs will need to understand the code as they work on it from time to time.  In the test suite I don't see it as a blocker

**Review by dheitzer — APPROVED** [review]
Looking good.

File: front-end/cypress/integration/sprint-7-test-119.spec.ts (hunk 1 above)
Comment (Elaine-Krauss-TCG) [reply front-end/cypress/integration/sprint-7-test-119.spec.ts]: Thanks for pointing this out.  It was made obsolete by a pass updating many for-loops to for-of-loops, and I missed the variable declaration there.  It's been stripped out now.

File: front-end/cypress/integration/sprint-7-test-192,248,249,250.spec.ts (hunk 2 above)
Comment (Elaine-Krauss-TCG) [reply front-end/cypress/integration/sprint-7-test-192,248,249,250.spec.ts]: Line removed

**Comment by Elaine-Krauss-TCG** [comment]
> 

Re: the "cognitive load" code smells.  There might be some merit when it comes to GenerateReportObject(), but the other "functions" flagged for "cognitive load" are the Cypress describe() wrappers that encompass pretty much entire e2e test files.  SonarCloud sees these as singular functions, but I don't think anyone digging around in them would see them as such.

File: front-end/cypress/integration/sprint-7-test-119.spec.ts [hunk 3]
```diff
@@ -98,7 +98,6 @@ describe('QA Test Script #119 (Sprint 7)', () => {
   it("Steps 4 & 5: Verify 'Candidate Office' dropdown contains House, Senate, and Presidential and then select Presidential", () => {
     cy.get(FieldCandidateOffice).click();
 
-    let OfficeType: string;
     const OfficeTypes: Array<string> = ['House', 'Senate', 'Presidential'];
```
Comment (toddlees) [inline front-end/cypress/integration/sprint-7-test-119.spec.ts]: Shoot i wish i caught this when i made the other comment.  The variable name should be camelCase: officeTypes and officeType

**Comment by Elaine-Krauss-TCG** [comment]
> 

New commit just pushed that reduces the code complexity of the GenerateReportObject() function.
At the moment, I have no intention of addressing code complexity within the Cypress describe() wrappers/

File: front-end/cypress/integration/sprint-7-test-192,248,249,250.spec.ts [hunk 4]
```diff
@@ -102,20 +102,19 @@ describe('QA Test Scripts #192, #248, #249, & #250 (Sprint 7)', () => {
   });
 
   for (ContactType of Object.keys(Contacts)) {
-    context(`---> ${ContactType}`, (c_type = ContactType) => {
-      let Contact: object = Contacts[c_type];
+    context(`---> ${ContactType}`, (CType = ContactType) => {
```
Comment (toddlees) [inline front-end/cypress/integration/sprint-7-test-192,248,249,250.spec.ts]: Sorry if this wasn't explained correctly.  the variable names should be camelCase.  So `ContactType` would become `contactType` and `Contacts` would become `contacts`

**Review by toddlees — CHANGES_REQUESTED** [review]
I know literally the last thing you want to do is go back and change variable names, but the variable names on typescript code should be camelCase with a lowercase first character.  Class names are CamelCase with an upper first character

**Comment by Elaine-Krauss-TCG** [comment]
> I know literally the last thing you want to do is go back and change variable names, but the variable names on typescript code should be camelCase with a lowercase first character. Class names are CamelCase with an upper first character

Oh heavens...  It probably should have gone unspoken, but since it wasn't mentioned [here](https://github.com/fecgov/fecfile-web-app/pull/273#pullrequestreview-974623753), I didn't know.

This will take a couple hours to incorporate, but it needs to be done...

**Comment by toddlees** [comment]
> > 
> 
> Re: the "cognitive load" code smells. There might be some merit when it comes to GenerateReportObject(), but the other "functions" flagged for "cognitive load" are the Cypress describe() wrappers that encompass pretty much entire e2e test files. SonarCloud sees these as singular functions, but I don't think anyone digging around in them would see them as such.

Like I said I don't think they should block this ticket, but what it's recommending is that we restructure the code such that someone reading the code does not need to keep track of many loop scopes in one function.  We can fix it by pulling portions of it out into functions that are called in the describe.  That way, when you're looking at the describe, you get a non-complex view of what's happening and if you need to work with a specific part, you can dive into the function that's abstracting it.

**Comment by toddlees** [comment]
> > > 
> > 
> > 
> > Re: the "cognitive load" code smells. There might be some merit when it comes to GenerateReportObject(), but the other "functions" flagged for "cognitive load" are the Cypress describe() wrappers that encompass pretty much entire e2e test files. SonarCloud sees these as singular functions, but I don't think anyone digging around in them would see them as such.
> 
> Like I said I don't think they should block this ticket, but what it's recommending is that we restructure the code such that someone reading the code does not need to keep track of many loop scopes in one function. We can fix it by pulling portions of it out into functions that are called in the describe. That way, when you're looking at the describe, you get a non-complex view of what's happening and if you need to work with a specific part, you can dive into the function that's abstracting it.

There's absolutely a precedent of ignoring the cognitive load smells in some utility scripts i personally made for the api and validate repos.  Just letting you know that in the core codebase, we are going to try much harder to avoid these.

**Comment by toddlees** [comment]
> > I know literally the last thing you want to do is go back and change variable names, but the variable names on typescript code should be camelCase with a lowercase first character. Class names are CamelCase with an upper first character
> 
> Oh heavens... It probably should have gone unspoken, but since it wasn't mentioned [here](https://github.com/fecgov/fecfile-web-app/pull/273#pullrequestreview-974623753), I didn't know.
> 
> This will take a couple hours to incorporate, but it needs to be done...

Oh i wish this happened on a smaller ticket 😭

**Comment by Elaine-Krauss-TCG** [comment]
> 

Oh sorry, I hope I didn't come off as snippy there.  What I intended to convey was that I fully agree that cognitive load code smells are something that we should work to avoid (hence the adjustments I made to GenerateReportObject in reports.spec.ts), and that my ignoring the three other cognitive load code smells was intentional.

**Comment by Elaine-Krauss-TCG** [comment]
> > > I know literally the last thing you want to do is go back and change variable names, but the variable names on typescript code should be camelCase with a lowercase first character. Class names are CamelCase with an upper first character
> > 
> > 
> > Oh heavens... It probably should have gone unspoken, but since it wasn't mentioned [here](https://github.com/fecgov/fecfile-web-app/pull/273#pullrequestreview-974623753), I didn't know.
> > This will take a couple hours to incorporate, but it needs to be done...
> 
> Oh i wish this happened on a smaller ticket sob

Well, at least I'll never forget the camelCase rules moving forward, hah

**Comment by Elaine-Krauss-TCG** [comment]
Updated everything from "CamelCase" to "camelCase"

**Review by toddlees — APPROVED** [review]

**Pull request merged**